src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -179,6 +180,11 @@ public void RegisterWorkspaceHost(IVisualStudioWorkspaceHost host)
             _workspaceHosts.Add(new WorkspaceHostState(this, host));
         }
 
+        public Task RegisterWorkspaceHostFromForegroundThreadAsync(IVisualStudioWorkspaceHost host)
```

Comment:
ping @jasonmalinowski is using Dispatcher is the way we should do? or should I get one from CurrentForegroundThreadData?